### ISSUE TITLE
🛣 Roadmap 🎯

### ISSUE THREAD

**Issue by sbernard31** [eclipse/wakaama#597]
The project probably need a kind of Roadmap.
Not necessary something very accurate but at least some hints for the community about the direction of the project.

You can have a look at what is [done for Leshan](https://github.com/eclipse/leshan/wiki/Roadmap) even if this is probably not the better example. 

I think this is probably linked in a way to those topics : 
 - the future of the 1.0 version. (#487)
 - Defining supported compilers, architectures, language versions (#494)

**Comment by mlasch** [comment]
Having a roadmap which leads to a Wakaama 2.0 (?) version would be very nice. We need to identify what is missing for LWM2M 1.0 and 1.1. and what we want to have in a future release. @qleisan you already started a list of [supported features](https://github.com/eclipse/wakaama/wiki/LWM2M-Supported-Features)? Having a dedicated roadmap wiki page like the one at Leshan may help. Simultaneously, we can remove the `TODO` file from the repository.

However, features we consider supported should have at least some basic tests (which integrates with the CI), so we can be sure to not break thinks. Some work in this regard is ongoing #598.

**Comment by qleisan** [comment]
I agree. Tests also help to clarify what is supported (partially or fully).

**Comment by rettichschnidi** [comment]
For any subsequent release:
* definitely more testing than what we have currently
* Proper developer documentation (neither the current README.md nor the GitHub-Bound Wiki is well-enough solution): #586
* Copyright Headers: SPDX #584

For a 2.0 release:
* A better API: #544
* Switch to main :)
* All findings by code analyzing tools dealt with (Clang static analyzer, SonarQube)

**Comment by dongfuhai** [comment]
When will the next official version of wakaama be released?

**Comment by tuve** [comment]
Actually there is no release plan as features are implemented on a voluntary basis as needed. That said I think we should perhaps do a a new release since quite a few new features has been added since the last release. What do you think @rettichschnidi, @mlasch

**Comment by LukasWoodtli** [comment]
Closing because the Wakaama project will be terminated: See issue #883 and [iot.wakaama 2026.05 Termination Review](https://gitlab.eclipse.org/eclipsefdn/emo-team/emo/-/issues/1191)
